examples/fixtures/sqlx/main.go

```diff
@@ -61,14 +62,20 @@ func main() {
 				return nil, err
 			}
 
-			return func(ctx context.Context) error {
+			return func(ctx context.Context) (err error) {
 				var rows *sqlx.Rows
 				user := &User{}
 
 				if rows, err = client.Queryx(ctx, "SELECT * FROM users"); err != nil {
 					return err
 				}
 
+				defer func() {
+					if closeErr := rows.Close(); closeErr != nil {
+						err = multierror.Append(err, fmt.Errorf("closing rows failed: %w", closeErr))
+					}
+				}()
+
 				for rows.Next() {
```

Comment:
Would be cool if we could simultaneously tackle checking rows.Err() after for loops iterating rows.Next()